hpx/runtime/threads/detail/create_thread.hpp

```diff
@@ -77,7 +77,7 @@ namespace hpx { namespace threads { namespace detail
         {
             if (data.priority == thread_priority_default &&
                 thread_priority_high_recursive ==
-                    threads::get_self_id()->get_priority())
+                    get_thread_id_data(threads::get_self_id())->get_priority())
```

Comment:
We could even wrap this into a single function, something like `threads::get_self_data()` or similar. That would simplify writing the above to: `threads::get_self_data()->get_priority()`.

Reply:
Doh. Good idea!

Reply:
I've added a `threads::get_self_id_data` (I left `id` in there to indicate that it comes from the thread id). `get_thread_id_data` is still there as well for the cases when one gets a thread id from elsewhere.